app/src/webkit/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -67,6 +70,26 @@ private static void configureSettings(Context context, WebSettings settings) {
         settings.setAllowFileAccess(false);
 
         settings.setBlockNetworkImage(appSettings.shouldBlockImages());
+
+        appSettings.subscribeToBlockImagesPreference(new Settings.BooleanPrefListener() {
+            final WeakReference<WebView> webViewWeakReference = new WeakReference<>(webView);
```

Comment:
Even though you are using a WeakReference here there's still a implicit reference to the final WebView and Context objects from the parent class.

But even with the WeakReference working this might create a bunch of listeners that don't do anything and do not get cleaned up. Maybe this can move inside WebkitView: When the view is attached it registers a listener, updates its setting on changes and unregisters if it is being detached.